apix-impl/src/main/java/eu/xenit/apix/alfresco/metadata/NodeService.java

```diff
@@ -373,54 +374,58 @@ private boolean containsAspect(List<AspectDefinition> aspects, AspectDefinition
 
     @Override
     public List<ChildParentAssociation> getChildAssociations(eu.xenit.apix.data.NodeRef ref) {
-        List<ChildParentAssociation> apixChildAssocs = new ArrayList<>();
-        List<ChildAssociationRef> alfChildAssocs = this.nodeService.getChildAssocs(c.alfresco(ref));
-        for (ChildAssociationRef alfChildAssoc : alfChildAssocs) {
-            NodeRef alfChildRef = alfChildAssoc.getChildRef();
-            QName alfType = alfChildAssoc.getTypeQName();
-            boolean isPrimary = alfChildAssoc.isPrimary();
-            ChildParentAssociation apixChildAssoc = new ChildParentAssociation(ref, c.apix(alfChildRef),
-                    c.apix(alfType), isPrimary);
-            apixChildAssocs.add(apixChildAssoc);
-        }
-
-        return apixChildAssocs;
+        return nodeService.getChildAssocs(c.alfresco(ref))
+                .stream().map(alfChildAssoc ->
+                        new ChildParentAssociation(
+                                ref,
```

Comment:
@WimCrols I want to note that there might be a semantic error here (also in the original) in that both getParent and getChild return association objects in which the target noderef is listed as the source of the returned association. This means that in one case (imo the getChild) the direction of the returned association is inverted?
Maybe we should open another ticket for it?

Reply:
Good idea to open another ticket.